modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -209,3 +225,33 @@ def row_apply_func(row):
         isinstance(ptn, new_large_modin_frame._partition_mgr_cls._partition_class)
         for ptn in new_large_modin_frame._partitions.flatten()
     ), "Partitions are not block partitioned after apply."
+
+
+@pytest.mark.skipif(
+    Engine.get() not in ("Dask", "Ray"),
+    reason="Only Dask and Ray engines have virtual partitions.",
+)
+def test_making_virtual_partition_out_of_virtual_partitions_with_call_queue():
```

Comment:
My undestanding is this test creates two virtual partitions, each with calls in their call queues. the `level_two_virtual` is the composed of the `level_one_virtual` partition. This PR makes it so that when level two is drained, level one is also drained?

If so, my question is are there other cases worth testing here, such as 3-level (perhaps no different than the current one because it's just more composition), multiple virtual inheritance (level 2 composed of two level 1 virtual partitions), circular inheritance (level 2 composed of level 1, then level 1 composed of 2).

Reply:
> This PR makes it so that when level two is drained, level one is also drained?

That's correct. Before this PR, we just drop the level one call queue on the floor.

> If so, my question is are there other cases worth testing here

I added tests for the 3-level virtual partition composition, multiple virtual inheritance, and virtual partition composed of both virtual and block partition.